autovalet_parking/src/autovalet_parking/Parker.py

```diff
@@ -223,29 +219,22 @@ def calculate_error(self):
         trans_err = (self.target_pose[:-1] - self.reached_pose[:-1])*100
         # Error in orientation (yaw) in degrees
         # make sure the orientation is the same in hardware so that the "270" works as well
-
-        align_tag_to_goal = 0
-        if self.park_direction == "left":
-            align_tag_to_goal = -90
-        elif self.park_direction == "right":
-            align_tag_to_goal = 90
+        align_tag_to_goal = -90 if self.park_direction == "left" else 90
         yaw_err   = np.abs((self.target_pose[2]*180/3.14 + align_tag_to_goal) - self.reached_pose[2]*180/3.14) #comparing the aruco tag (rotated by 270deg) with husky's yaw
-        if yaw_err>180:
-                yaw_err -=180
-        if yaw_err > 360:
+        print 'yaw err b4 wrapping', yaw_err
+        if yaw_err > 180:
+            yaw_err -= 180
+        if yaw_err > 300:
             yaw_err -= 360
-
         print "Target pose (x,y,theta): ", self.target_pose[:-1]*100, self.target_pose[2]*180/3.14 + align_tag_to_goal
         print "Reached pose (x,y,theta): ", self.reached_pose[:-1]*100, self.reached_pose[2]*180/3.14
         print "Translation error (cms): ", trans_err
-        print "================================"
-        print "Net error (norm of the error)", np.linalg.norm(trans_err)
         print "Orientation error (deg): ", yaw_err
+        print "Net error (norm of the error)", np.linalg.norm(trans_err)
 
     def get_target_pose(self, gt):
-        for index, name in enumerate(gt.name):
-            if name == "aruco_visual_marker_7::marker":
-                self.aruco_tag_gt    = gt.pose[index]
+
+        self.aruco_tag_gt    = gt.pose[-18] # name 'aruco_visual_marker_7::marker' <to-do> find the name! Don't HC
```

Comment:
lmao `todo`

yea i don't wanna do that

Reply:
yah I don't even understand what's happening. will just delete everything I don't understand.

Reply:
LOL